src/ephy-bookmark.c

```diff
@@ -19,16 +19,31 @@
 
 #include "ephy-bookmark.h"
 
+#include "ephy-shell.h"
+#include "ephy-sync-crypto.h"
+#include "ephy-sync-utils.h"
+
+#include <string.h>
+
 struct _EphyBookmark {
   GObject      parent_instance;
 
   char        *url;
   char        *title;
   GSequence   *tags;
   gint64       time_added;
+
+  char        *id;
+  double       modified;
+  gboolean     uploaded;
 };
 
-G_DEFINE_TYPE (EphyBookmark, ephy_bookmark, G_TYPE_OBJECT)
+static JsonSerializableIface *serializable_iface = NULL;
+
+static void json_serializable_iface_init (gpointer g_iface);
+
+G_DEFINE_TYPE_EXTENDED (EphyBookmark, ephy_bookmark, G_TYPE_OBJECT, 0,
```

Comment:
I didn't know that this macro existed. You should be able to use `G_DEFINE_TYPE_WITH_CODE` instead, which is more idiomatic.